Fix servo HTTP metrics.

NiteshKant opened the pull request:
The HttpClientListener and HttpServerListener were listening for HttpClient/ServerMetricEvents instead of Client/ServerMetricEvents.
This causes ClassCastExceptions while receiving Client/ServerMetricEvents.

The test did not catch this as the listener in the test was a TCP listener attached to an HttpClient. Since all Http*MetricEvents can be converted to *MetricEvents and not the other way around, the tests were happy.

Commit:
Fix servo HTTP metrics.

The HttpClientListener and HttpServerListener were listening for HttpClient/ServerMetricEvents instead of Client/ServerMetricEvents.
This causes ClassCastExceptions while receiving Client/ServerMetricEvents.

The test did not catch this as the listener in the test was a TCP listener attached to an HttpClient. Since all Http*MetricEvents can be converted to *MetricEvents and not the other way around, the tests were happy.

Commit:
Optimized imports

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #93](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/93/) FAILURE
Looks like there's a problem with this pull request

Commit:
Fix for servo metrics (contd)

Now added the servo listeners to the examples which also covers the basic sanity of these event types.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #94](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/94/) SUCCESS
This pull request looks good
